src/DataStructures/Mesh.hpp

```diff
@@ -32,16 +33,19 @@ enum class Quadrature;
  * placement of grid points in the computational domain. It does so through a
  * choice of basis functions, quadrature and number of points \f$N\f$ in each
  * dimension. The grid points are the associated collocation points and can be
- * computed by `Spectral::collocation_points`. This means that a simulated
- * physical field can be represented by its value on each grid point and then
- * approximated by a polynomial of degree \f$p=N-1\f$ through a linear
- * combination of Lagrange polynomials.
+ * obtained from Spectral::collocation_points(const Mesh<1>&):
  *
- * \note A field represented by a `DataVector` has no meaning without an
- * accompanying `Mesh` to provide context. Only w.r.t a `Mesh` can a
- * `DataVector` of length `mesh.number_of_grid_points()` be interpreted as the
- * field values at the collocation points. These field values are identical to
- * the nodal expansion coefficients in Lagrange polynomials.
+ * \snippet Test_Spectral.cpp get_points_for_mesh
+ *
+ * A simulated physical field can be represented by a DataVector of length
+ * number_of_grid_points() that holds the field value on each point of
+ * the computational grid. These values are identical to the field's nodal
+ * expansion coefficients. They approximate the field by a polynomial of degree
+ * \f$p=N-1\f$ through a linear combination of Lagrange polynomials.
+ *
+ * \note A DataVector has no meaning without its accompanying Mesh to
```

Comment:
there are uses for a DataVector without a Mesh, for example any arbitrary set of points, in which case you wouldn't need a Mesh, but corresponding coordinate values (most likely stored in a Tensor<DataVector> itself).  So I would either remove this note or soften it.  Also there are many functions that just do point-wise operations where you don't care about the Mesh or an associated set of coordinate values

Reply:
Makes sense, removed that note 👍